### ISSUE TITLE
Issue Launching with OptiFine

### ISSUE THREAD

**Issue by Fuldmathr** [MultiMC/MultiMC5#2524]
System Information
-----------------------------
MultiMC version: 0.6.4-1291

Operating System: Windows 10 64-bit

Summary of the issue or suggestion:
----------------------------------------------
When I attempt to launch an instance of 1.13.2 with OptiFine, I get an error saying that some artifacts marked as 'local' are missing their files.

What should happen:
------------------------------
The game launches.

Suspected cause:
---------------------------
I must have done something wrong in setting it up, but I followed the page https://github.com/MultiMC/MultiMC5/wiki/MultiMC-and-OptiFine and can't find any error in the Json file, file location, etc.

**Logs/Screenshots:**
---------------------------
MultiMC version: 0.6.4-1291

Minecraft folder is:
C:/Users/Furui/Downloads/mmc-stable-win32/MultiMC/instances/1.13.2/.minecraft

Java path is:
C:/Program Files (x86)/Common Files/Oracle/Java/javapath/javaw.exe

Java is version 1.8.0_201, using 64-bit architecture.

Instance update failed because: Some artifacts marked as 'local' are missing their files:
C:/Users/Furui/Downloads/mmc-stable-win32/MultiMC/instances/1.13.2/libraries/OptiFine_1.13.2_HD_U_E5.jar

You need to either add the files, or removed the packages that require them.
You'll have to correct this problem manually.

json contents
---------------------------
{
    "+tweakers": [
        "optifine.OptiFineTweaker"
    ],
    "formatVersion": 1,
    "libraries": [
        {
            "name": "net.minecraft:launchwrapper:1.12"
        },
        {
            "MMC-filename": "OptiFine_1.13.2_HD_U_E5.jar",
            "MMC-hint": "local",
            "name": "optifine:OptiFine:1.13.2_HD_U_E5"
        }
    ],
    "mainClass": "net.minecraft.launchwrapper.Launch",
    "name": "OptiFine",
    "requires": [
        {
            "equals": "1.13.2",
            "uid": "net.minecraft"
        }
    ],
    "uid": "optifine.Optifine",
    "version": "1.13.2_HD_U_E5"
}

----------------------------
[//]: # (Please refer to https://github.com/MultiMC/MultiMC5/wiki/Log-Upload for instructions on how to attach your logs.)

Additional Info:
---------------------------
I enter the json file just as the example shows, switching OptiFine_1.13.2_HD_U_E4 out of OptiFine_1.13.2_HD_U_E5

Sorry if this should be simple, thanks for your time.

**Comment by peterix** [comment]
Well, you need to add the `C:/Users/Furui/Downloads/mmc-stable-win32/MultiMC/instances/1.13.2/libraries/OptiFine_1.13.2_HD_U_E5.jar` file.

**Comment by Fuldmathr** [comment]
![mmcopti](https://user-images.githubusercontent.com/47190040/52011456-06df9d80-24a6-11e9-8633-b1f3b5d1f4b7.PNG)

**Comment by peterix** [comment]
delete that json and remove that extra folder the jar is in

**Comment by peterix** [comment]
Unless I'm looking at this wrong. That UI is confusing.

**Comment by Fuldmathr** [comment]
Still the same error

![error](https://user-images.githubusercontent.com/47190040/52011814-fd0a6a00-24a6-11e9-9f89-03c8e5be9347.PNG)

**Comment by peterix** [comment]
You need `OptiFine_1.13.2_HD_U_E5.jar` to be in `C:/Users/Furui/Downloads/mmc-stable-win32/MultiMC/instances/1.13.2/libraries/`. Don't put it in more folders, don't assume it will search for it like you did there. That's all there is to it.

**Comment by Fuldmathr** [comment]
Yes the executable jar file is in libraries now without the extra folder like you said in your last comment, but I still get the same error.

**Comment by peterix** [comment]
Then it is not there. Maybe yoiu put something else there, that simply does not belong there at all?

Now that I look at it, it does seem like you did something weird with optifine and turned it into a version jar along with the json. Don't do that. Just take the original jar and put it in the folder.

**Comment by Fuldmathr** [comment]
That was the problem! The reason it was weird is because for the regular Minecraft launcher, you run OptiFine and turn it into that jar along with the json so I figured it was the same. Thank you.

**Comment by Fuldmathr** [comment]
Hm, actually after doing that it launches perfectly fine but OptiFine just isn't running.

**Comment by peterix** [comment]
Full log? Use the upload button and link it here.

**Comment by Fuldmathr** [comment]
https://paste.ee/p/8kGdZ

**Comment by peterix** [comment]
That's strange.

If you are using the json you posted, the main class should be from launchwrapper, not Minecraft.

json:
```
"mainClass": "net.minecraft.launchwrapper.Launch",
```

log:
```
Main Class:
  net.minecraft.client.main.Main
```

**Comment by Fuldmathr** [comment]
Okay, I completely uninstalled and reinstalled this time I believe doing it correctly, and the main class is now from launchwrapper however I still have the same issue.

https://paste.ee/p/d4KIi

**Comment by peterix** [comment]
That looks like a very different issue. It clearly works now, from the launcher's point of view. OptiFine does things and then fails to do things for no apparent reason:

```
[08:31:57] [main/INFO]: Created: 1024x512 textures-atlas
[08:31:57] [main/INFO]: [OptiFine] Animated sprites: 27
[08:31:59] [main/INFO]: Narrator library for x64 successfully loaded
[08:31:59] [main/INFO]: [OptiFine] *** Reloading custom textures ***
[08:32:07] [main/INFO]: Stopping!
[08:32:08] [main/INFO]: SoundSystem shutting down...
[08:32:08] [main/INFO]: SoundSystem Author: Paul Lamb, www.paulscode.com
AL lib: (EE) alc_cleanup: 1 device not closed
Process exited with code 0.
```

**Comment by peterix** [comment]
Just to confirm, can you try importing this instance and launching it?
It should end up with the same problem.

Just drag the file into the main window. It has everything pre-installed.

[OptiFine_1.13.2_HD_U_E5.zip](https://github.com/MultiMC/MultiMC5/files/2817506/OptiFine_1.13.2_HD_U_E5.zip)

**Comment by peterix** [comment]
Try to raise the amount of memory given to MC to 2GB.

**Comment by Fuldmathr** [comment]
Ah I see, Minecraft normally uses 2GB for me already that's why I don't have issues. Thank you.

**Comment by peterix** [comment]
That gives me some ideas...

The Minecraft versions should have suggested memory sizes.
